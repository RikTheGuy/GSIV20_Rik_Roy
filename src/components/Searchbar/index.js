import React from 'react'

import './index.scss'
import 'material-icons/iconfont/filled.css'

const index = ({ query = '', setQuery, onSearch, onCancel }) => {
    return (
        <form data-testid='searchbar' className='searchbar' onSubmit={onSearch}>
            <span className='material-icons icon'>search</span>
            <input id='search-input' type='text' placeholder='Search' value={query} onChange={e => setQuery(e.target.value)} />
            <div className='buttons'>
                <span className='material-icons icon' onClick={onCancel}>clear</span>
                {
                    query.length > 0 &&
                    <span className='material-icons icon' onClick={onSearch}>search</span>
                }
            </div>
        </form>
    )
}

export default index
