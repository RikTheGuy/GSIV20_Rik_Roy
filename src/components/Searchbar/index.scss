@import "../../styles/colors.scss";

.searchbar {
  display: flex;
  flex: 1;
  flex-direction: row;
  align-items: center;
  border: 1px solid $disabled;
  border-radius: 0.3rem;
  padding: 0.1rem 0.3rem;
  background-color: $disabled;
  color: $light-gray;

  input#search-input {
    border: 1px solid transparent;
    width: 100%;
    font-size: 1rem;
    background-color: $disabled;
    color: $gray;

    &::placeholder {
      color: $light-gray;
    }
    &:focus {
      outline: none;
      border: 1px solid transparent;
    }
  }

  .icon {
    font-size: 1.2rem;
  }

  .buttons {
    display: flex;
    justify-content: row;
    cursor: pointer;
  }
}
