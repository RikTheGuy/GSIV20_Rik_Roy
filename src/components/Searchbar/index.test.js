import Searchbar from './index.js';
import { render, cleanup } from '@testing-library/react';

import renderer from 'react-test-renderer';

var query = ''

describe('Searchbar', () => {

    afterEach(cleanup)

    describe('Renders Successfully', () => {
        test('Runs without crashing', () => {
            render(<Searchbar />);
        })
    })

    describe('Snapshot Testing', () => {
        test('Renders Properly', () => {
            const tree = renderer.create(<Searchbar />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})