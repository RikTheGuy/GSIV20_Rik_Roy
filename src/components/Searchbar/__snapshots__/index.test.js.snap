// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar Snapshot Testing Renders Properly 1`] = `
<form
  className="searchbar"
  data-testid="searchbar"
>
  <span
    className="material-icons icon"
  >
    search
  </span>
  <input
    id="search-input"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <div
    className="buttons"
  >
    <span
      className="material-icons icon"
    >
      clear
    </span>
  </div>
</form>
`;
